WordPress/gutenberg
# Categories: Add a list style type option to the categories block

## What?
This adds a list style type attribute to the categories list block.

## Why?
The current block isn't customizable, so it's not possible to change the discs without CSS. This is quite a strong opinion, so it would be good if themes could use the block without needing to use this style.

## How?
Adds a `listStyleType` attribute to the categories block.

## Testing Instructions
1. Add some categories to your site
2. Add a category block to a post
3. Open the inspector controls for the block
4. Check that the value for "List Style" is default
5. Toggle on the "Display as dropdown" option
6. Check that the "List Style" option no longer shows
7. Change the List Style attribute to a different value.
8. Check that the categories list now uses the selected list style in the editor
9. Save your change
10. Check that the categories list now uses the selected list style in the frontend

## Screenshots or screencast 
<img width="244" alt="Screenshot 2023-06-08 at 16 58 04" src="https://github.com/WordPress/gutenberg/assets/275961/1c3e891a-8694-4616-b0c3-08b541a7a1d3">
